packages/block-editor/src/components/block-draggable/use-scroll-when-dragging.native.js

```diff
@@ -0,0 +1,135 @@
+/**
+ * External dependencies
+ */
+import { Dimensions } from 'react-native';
+import {
+	useSharedValue,
+	useAnimatedRef,
+	scrollTo,
+	useAnimatedReaction,
+	withTiming,
+	withRepeat,
+	cancelAnimation,
+	Easing,
+} from 'react-native-reanimated';
+
+/**
+ * Internal dependencies
+ */
+import { useBlockListContext } from '../block-list/block-list-context';
+
+const SCROLL_INACTIVE_DISTANCE_PX = 50;
+const SCROLL_INTERVAL_MS = 1000;
+const VELOCITY_MULTIPLIER = 5000;
+
+/**
+ * React hook that scrolls the scroll container when a block is being dragged.
+ *
+ * @return {Function[]} `startScrolling`, `scrollOnDragOver`, `stopScrolling`
+ *                      functions to be called in `onDragStart`, `onDragOver`
+ *                      and `onDragEnd` events respectively. Additionally,
+ * 						`scrollHandler` function is returned which should be
+ * 						called in the `onScroll` event of the block list.
+ */
+export default function useScrollWhenDragging() {
+	const { scrollRef } = useBlockListContext();
+	const animatedScrollRef = useAnimatedRef();
+	animatedScrollRef( scrollRef );
+
+	const windowHeight = Dimensions.get( 'window' ).height;
```

Comment:
Alternatively, we could use `useWindowDimensions` which updates automatically if the window size changes. What do you think?

Reply:
Good point, I'll replace this with `useWindowDimensions` 👍 .

Reply:
I've applied this suggestion in https://github.com/WordPress/gutenberg/pull/39705/commits/440217c70c434c784b57bf1312c95f0c43529079.